examples/swash_render/src/main.rs

```diff
@@ -186,8 +186,14 @@ fn main() {
                 PositionedLayoutItem::InlineBox(inline_box) => {
                     for x_off in 0..(inline_box.width.floor() as u32) {
                         for y_off in 0..(inline_box.height.floor() as u32) {
+                            // Round the baseline to pixel boundary.
+                            // An even better renderer would account for fractional DPI scale.
+                            // To get the baseline we need to calculate the bottom edge of the box.
```

Comment:
Aren't these metrics already in physical pixels?

Reply:
They are, but in `f32` with fractional bits. So it'll get spread across multiple device pixels unless we round to where the device pixel boundary is.